pip/python-daemon.file

```diff
@@ -1 +1,2 @@
 Requires: py3-docutils py3-lockfile
+%define source0 https://pagure.io/python-daemon/archive/release/3.0.2/python-daemon-release/3.0.2.tar.gz
```

Comment:
@iarspider , looks like this is not correct. this source does not exist any more.
I was able to build `python-daemon 3.0.2` by
- removing the explicit source here
- updating pip/requirements.txt to use correct version `3.0.2`
- and I think a had to add dependency on `py3-packaging`

But then `py3-luigi`  failed with error `ModuleNotFoundError: No module named 'util'`

Reply:
![image](https://github.com/user-attachments/assets/4aa6f5d7-dbd0-4a4a-85e8-d7648af7fce3)
3.0.2 is marked as "yanked" on PyPI, probably that is the reason why luigi fails

Reply:
did you try building `3.0.1`  ( note that for 3.0.1 you need to remove the explicit `%define source0` otherwise it will still use hard coded 3.0.2 sources

Reply:
3.0.1 is what we currently have, and it fails to import `setuptools.extern`:

```
Processing ./src.tar.gz
  Preparing metadata (pyproject.toml): started
  Running command Preparing metadata (pyproject.toml)
  Traceback (most recent call last):
    File "/build/razumov/cmsbuild/BUILD/el8_amd64_gcc12/external/py3-pip/24.0-b1aa17d03e426f17ac26e10ec6f018b6/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
      main()
    File "/build/razumov/cmsbuild/BUILD/el8_amd64_gcc12/external/py3-pip/24.0-b1aa17d03e426f17ac26e10ec6f018b6/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/build/razumov/cmsbuild/BUILD/el8_amd64_gcc12/external/py3-pip/24.0-b1aa17d03e426f17ac26e10ec6f018b6/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 149, in prepare_metadata_for_build_wheel
      return hook(metadata_directory, config_settings)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/build/razumov/cmsbuild/BUILD/el8_amd64_gcc12/external/py3-setuptools/72.1.0-f09b74ebddf75c63bf10f1dadf250d18/lib/python3.12/site-packages/setuptools/build_meta.py", line 368, in prepare_metadata_for_build_wheel
      self.run_setup()
    File "/build/razumov/cmsbuild/BUILD/el8_amd64_gcc12/external/py3-setuptools/72.1.0-f09b74ebddf75c63bf10f1dadf250d18/lib/python3.12/site-packages/setuptools/build_meta.py", line 313, in run_setup
      exec(code, locals())
    File "<string>", line 23, in <module>
    File "/build/razumov/cmsbuild/BUILD/BUILD/el8_amd64_gcc12/external/py3-python-daemon/3.0.1-73c71b7d6919283fd1c0d9f34dc746ed/cmsdist-tmp/pip-req-build-vmyvfrut/version.py", line 39, in <module>
      import setuptools.extern.packaging.version
  ModuleNotFoundError: No module named 'setuptools.extern'
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
```

The author of the `python-daemon` package is trying to do some really weird tricks with setuptools using undocumented functionality.

Reply:
Also see https://pagure.io/python-daemon/issue/94